mlrun/model_monitoring/db/tsdb/tdengine/schemas.py

```diff
@@ -0,0 +1,199 @@
+    Metrics, and Predictions.
+    """
+
+    def __init__(
+        self,
+        super_table: str,
+        columns: dict[str, str],
+        tags: dict[str, str],
+    ):
+        self.super_table = super_table
+        self.columns = columns
+        self.tags = tags
+        self.database = _MODEL_MONITORING_DATABASE
+
+    def _create_super_table_query(self) -> str:
+        columns = ", ".join(f"{col} {val}" for col, val in self.columns.items())
+        tags = ", ".join(f"{col} {val}" for col, val in self.tags.items())
+        return f"CREATE STABLE if not exists {self.database}.{self.super_table} ({columns}) TAGS ({tags});"
+
+    def _create_subtable_query(
+        self,
+        subtable: str,
+        values: dict[str, str],
+    ) -> str:
+        values = ", ".join(f"'{values[val]}'" for val in self.tags)
+        return f"CREATE TABLE if not exists {self.database}.{subtable} using {self.super_table} TAGS ({values});"
+
+    def _insert_subtable_query(
+        self,
+        subtable: str,
+        values: dict[str, str],
+    ) -> str:
+        values = ", ".join(f"'{values[val]}'" for val in self.columns)
+        return f"INSERT INTO {self.database}.{subtable} VALUES ({values});"
+
+    def _delete_subtable_query(
+        self,
+        subtable: str,
+        values: dict[str, str],
+    ) -> str:
+        values = " AND ".join(
+            f"{val} like '{values[val]}'" for val in self.tags if val in values
+        )
+        if not values:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"values must contain at least one tag: {self.tags.keys()}"
+            )
+        return f"DELETE FROM {self.database}.{subtable} WHERE {values};"
+
+    def _drop_subtable_query(
+        self,
+        subtable: str,
+        # database: str = _MODEL_MONITORING_DATABASE
+    ) -> str:
+        return f"DROP TABLE if exists {self.database}.{subtable};"
+
+    def _get_subtables_query(
+        self,
+        values: dict[str, str],
+    ) -> str:
+        values = " AND ".join(
+            f"{val} like '{values[val]}'" for val in self.tags if val in values
+        )
+        if not values:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"values must contain at least one tag: {self.tags.keys()}"
+            )
+        return f"SELECT tbname FROM {self.database}.{self.super_table} where {values};"
+
+    @staticmethod
+    def _get_records_query(
+        table: str,
+        columns_to_filter: list[str] = None,
+        filter_query: str = "",
+        start: str = datetime.datetime.now().astimezone() - datetime.timedelta(hours=1),
+        end: str = datetime.datetime.now().astimezone(),
+        timestamp_column: str = "time",
+        database: str = _MODEL_MONITORING_DATABASE,
+    ) -> str:
+        full_query = "select "
+        if columns_to_filter:
+            full_query += ", ".join(columns_to_filter)
```

Comment:
1. The defaults of the `start` and `end` times are not what you probably expect. They are the "now" time when running this class/signature the first time, which is the import time.
2. If "now" datetime is needed, please use `mlrun.utils.datetime_now()`.
3. Don't use `+=` on strings, as it's not performant.
4. This method can (and should) be unit-tested easily.

Reply:
As we agreed:
-> default time values were removed
-> the query build is not using `StringIO()` instead of `+=`
-> unittests were added